core/src/main/scala/com/nvidia/spark/rapids/tool/planparser/PythonMapInArrowExecParser.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.tool.planparser
+
+import com.nvidia.spark.rapids.tool.qualification.PluginTypeChecker
+
+import org.apache.spark.sql.execution.ui.SparkPlanGraphNode
+
+case class PythonMapInArrowExecParser(
+    node: SparkPlanGraphNode,
+    checker: PluginTypeChecker,
+    sqlID: Long) extends ExecParser {
+
+  val fullExecName: String = node.name + "Exec"
+
+  override def parse: ExecInfo = {
+    // PythonMapInArrow doesn't have duration
+    val duration = None
+    val (speedupFactor, isSupported) = if (checker.isExecSupported(fullExecName)) {
+      (checker.getSpeedupFactor(fullExecName), true)
+    } else {
+      (1.0, false)
+    }
+    // TODO - add in parsing expressions - average speedup across?
+    ExecInfo(node, sqlID, node.name, "", speedupFactor, duration, node.id, isSupported, None)
```

Comment:
We need to verify the flags that will be applied on this exec.
For example, Do we need to make a change in `skipUDFCheckExecs` or `dataSetOrRDDRegExLookup` to include that Exec?
CC: @nartal1

Reply:
Thanks @amahussein  for this catch. I think there might be an issue with the way we skip UDF checks.

The current implementation within the SQLPlanParser.scala file uses `OR` condition to skip UDFs.

https://github.com/NVIDIA/spark-rapids-tools/blob/62ae855d50e5ddd937435ed6c2828c393c28d44a/core/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala#L295-L296

So if we return `ExecInfo(...,...,.., udf = true)` from  `PythonMapInArrow.parse()`, the above condition will be `true` even if we add to skip list i.e `containsUDF` will be `true` always.

We should refactor it to `AND` so that `containsUDF` becomes `false` for the execs we want to skip.
```
val containsUDF = udf && ExecHelper.isUDF(node)
```


Now, it works so far because we are never using the argument `ExecInfo(...,...,.., udf = true)` (eg, `ArrowEvalPythonExecParser` etc)

Reply:
I see. In that case we do not need to modify the `skipUDFCheckExecs` because `isUDF()` checks for `Execs` having the string `UDF` in Exec name

Reply:
Yes.
I think your point is legit from a different perspective:
In general, UDF is not supported. so, `ExecHelper.isUDF` does actually decide if this exec is supported or not.
Your point would be valid if we decide to break down the functionality so that we have "`isUDF`" Vs. "`isUDFSupported()`".
This would require a change in the fields of the ExecInfo. A refactor that does not seem urgent or necessary for now.
The point I say it is not necessary here is that I have an eye on supporting expression "PythonUDAF" which requires revisiting how we handle UDF functions anyway.

Reply:
Yes, that makes sense. Thank you Ahmed.